src/Symfony/Component/Process/Process.php

```diff
@@ -269,7 +269,8 @@ public function start(callable $callback = null, array $env = array())
                 // exec is mandatory to deal with sending a signal to the process
                 $commandline = 'exec '.$commandline;
             }
-        } else {
+        } elseif ('!' === substr($commandline = trim($commandline), 0, 1)) {
```

Comment:
why another magic thing instead of having a real api?

Reply:
I agree with @Tobion again. Although I don't think that `!` is magic, it's another "Symfony thing" that users must learn about. It'd be better to add some builder process. I hope Tobias or any other contributor can take over this. Thanks!

Reply:
another API is also something people will need to learn, there is little difference, except verbosity maybe
but anyway, your topic now :)